Added new terminology topic, and updated guidelines topic

CowboysFan opened the pull request:
@nhr @danmcp I'm adding a terminology doc to the 'contributing' dir as per our discussion. I've added some basic information, but the idea is to collaborate and finalize the use of our terminology for consistency. Please let me know what you think.

nhr commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -63,6 +63,25 @@ If using section anchors, they must be all lowercase letters, with no line space
+
+----
+Rollbacks revert an application back to a previous deployment and can be performed using the REST API or
+the OpenShift CLI. See the link:cli.html#deployment-rollbacks[CLI documentation] for more details.
```
I'm confused - your "don't do it this way" example is functionally identical to the "do it this way" example.

I think the "don't do it this way" example that you want is:

    See the http://docs.openshift.org/latest/using_openshift/cli.html#deployment-rollbacks[Openshift CLI]

CowboysFan replied:
@nhr I see where the confusion may be, and my bad for not clarifying. The examples don't show the method or markup of "creating" the link. What I was trying to show was to avoid linking by creating a separate sentence that begins with something like "See [this] for more information on x". Hopefully I've clarified it with my updates.

nhr commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -309,6 +328,36 @@ docker-registry     <none>                                    name=registrypod
 
+==== Quick reference
+.User accounts and info
+
```
Remove this whitespace between figure title and figure block.

CowboysFan replied:
@nhr Done, if that's what you were talking about.

CowboysFan commented:
@nhr Updates made. Please review.

adellape commented:
@CowboysFan Dunno if this is the best forum to bring it up (could just be added in a later PR, of course), but, can we get "Dockerfile" added to `term_glossary.adoc`? And a consensus on usage. Specifically:

- Should we mark it up using `[filename]`? Thinking so, since it is a file name. And if that's the case...
- Usually we would refer to a file as "the _filename_ file", but in this case that sounds redundant (i.e., "the Dockerfile file"). So should it just be referred to as "a/the Dockerfile"? See usage here: https://docs.docker.com/reference/builder/

Also side note, currently the "Overview" and intro para to "Usage of OpenShift Terms" are redundant.

nhr commented:
Corrections look good. Content is fine by me. Leaving this open for @CowboysFan to respond to @adellape 's questions.

CowboysFan commented:
@nhr @adellape I've updated the doc, again. Fixed:
* Overview and Usage of OpenShift Terms were redundant
* Added Dockerfile as a starting point

Also, I've created this doc just to get us started. The idea that each time some term or terminology issue arises, a writer or dev submits a PR to update this doc, a discussion takes place resulting in a final decision on the usage of that term or terminology. 

So I'm going to merge this for now, and if you or anyone else wants to add/update the terms, let's create a PR and open the discussion.
